cxplat/src/cxplat_winuser/fault_injection.cpp

```diff
@@ -36,6 +39,29 @@ typedef class _cxplat_fault_injection
     bool
     inject_fault();
 
+    /**
+     * @brief Reset the fault injection state, both in memory and on disk.
+     */
+    void
+    reset();
+
+    void
+    add_module_under_test(uintptr_t module_base_address, size_t module_size)
+    {
+        std::unique_lock lock(_mutex);
+        auto new_module = std::make_pair(module_base_address, module_base_address + module_size);
+        if (!_modules_under_test.contains(new_module)) {
+            _modules_under_test.insert(std::make_pair(module_base_address, module_base_address + module_size));
+        }
+    }
+
+    void
+    remove_module_under_test(uintptr_t module_base_address, size_t module_size)
+    {
```

Comment:
since add_module can be called multiple times on the same module, shouldn't this api support that too?  it doesn't look like it does now.